### PR TITLE
Added executable and library scripts and readme's

### PR DESCRIPTION
**Pull request by loekwous**

### CODE REVIEW

File: 7-ExecutablesAndLibraries/README.md [hunk 1]
```diff
@@ -1 +1,42 @@
...
+
+This chapter explains how to create executables and libraries. Executables and libraries are essential for managing projects
+with CMake.
+
+Executables and libraries are targets. A target is (just as in a Makefile) something that is going to be built, and which can be a required dependency
+for another target.
+
+# Executable
+
+An executable target is used to create for example a `.exe` file or for cross platform a `.elf`, `.hex`, or a `.elf`file.
```
Comment (MrFokker) [inline 7-ExecutablesAndLibraries/README.md]: An executable target generates files such as  `.exe` for Windows or cross-platform formats like `.bin`, `.elf` or `.hex`.

File: 7-ExecutablesAndLibraries/README.md [hunk 2]
```diff
@@ -1 +1,42 @@
...
+
+```CMake
+add_executable(target_name [sources])
+```
+
+The target name is required, whereas the sources are optional and can be specified later.
+
+# Libraries
+
+A library can be used to reuse compiled source files for different other libraries or executable.
```
Comment (MrFokker) [inline 7-ExecutablesAndLibraries/README.md]: "different other" the end of the line reads awkwardly

File: 7-ExecutablesAndLibraries/README.md [hunk 3]
```diff
@@ -1 +1,42 @@
...
+A minor nuance are the different type of libraries in CMake.
```
Comment (MrFokker) [inline 7-ExecutablesAndLibraries/README.md]: A minor nuance lies in the ....

File: 7-ExecutablesAndLibraries/CMakeLists.txt [hunk 4]
```diff
@@ -1,2 +1,11 @@
...
 project(MyFirstProject VERSION 0.1.0)
+
+
+# For linux platforms this executable is named `app`
+add_executable(app main.cpp)
+
+
+# For linux platforms this executable will create a `libmy_library.a`
+# Notice the lib prefix, which is prepended for linux platforms
+add_library(my_library STATIC my_library.cpp)
```
Comment (MrFokker) [inline 7-ExecutablesAndLibraries/CMakeLists.txt:11]: note: 'my_library' is made but not used. maybe add: `target_link_libraries(main PRIVATE my_library)`

**Review by MrFokker — COMMENTED** [review]
some comments that need to be loeked at. but looks good

File: 7-ExecutablesAndLibraries/CMakeLists.txt (hunk 4 above)
Comment (loekwous) [reply 7-ExecutablesAndLibraries/CMakeLists.txt:11]: I think that I want to save target manipulations for another chapter.

File: 7-ExecutablesAndLibraries/README.md (hunk 1 above)
Comment (loekwous) [reply 7-ExecutablesAndLibraries/README.md]: Fixed it

File: 7-ExecutablesAndLibraries/README.md (hunk 2 above)
Comment (loekwous) [reply 7-ExecutablesAndLibraries/README.md]: Fixed

File: 7-ExecutablesAndLibraries/README.md (hunk 3 above)
Comment (loekwous) [reply 7-ExecutablesAndLibraries/README.md]: Fixed

**Review by MrFokker — APPROVED** [review]
looks good to me

**Pull request merged**